apicast/src/resty/resolver.lua

```diff
@@ -251,6 +252,22 @@ local function search_dns(self, qname, stale)
   return answers, err
 end
 
+local function resolve_upstream(qname)
+  local peers, err = upstream.get_primary_peers(qname)
+
+  if not peers and err then
+    return nil, err
+  end
+
+  for i=1, #peers do
```

Comment:
According to the `if` above, here we could have `peers = nil`, and then `#peers` would raise an error.

Reply:
True that is possible. Will change the condition to return when there are no peers.